py_experimenter/experimenter.py

```diff
@@ -256,94 +265,140 @@ def fill_table_with_rows(self, rows: List[dict]) -> None:
         experiment is set to `done`. Otherwise, if an error occurred during the execution, the status is changed to 
-        `error` and the raised error is logged to the database table. 
+        `error` and the raised error is logged to the database table.
+        
+        Note that errors that occur before or after the execution of the `experiment_function` are logged via the function
+        `_handle_error`.
 
         :param experiment_function: The function that should be executed with the different parametrizations.
         :type experiment_function: Callable[[dict, dict, ResultProcessor], None]
-        :param max_experiments: The number of experiments to be executed by this `PyExperimenter`. If all experiments 
+        :param max_number_experiments_to_execute: The number of experiments to be executed by this `PyExperimenter`. If all experiments 
             should be executed, -1 can be used. Defaults to -1. 
-        :type max_experiments: int, optional
+        :type max_number_experiments_to_execute: int, optional
         :param random_order: Indicates whether the experiments to be executed are chosen consecutively by its experiment 
             ID (`False`) or in a randomized fashion (`True`). Defaults to False.
         :type random_order: bool, optional
         :raises InvalidValuesInConfiguration: If any value of the experiment parameters is of wrong data type.
         """
-        logging.info("Start execution of experiment_functions...")
-        keyfield_values = self.dbconnector.get_keyfield_values_to_execute()
-
-        if random_order:
-            shuffle(keyfield_values)
-
-        if 0 <= max_experiments < len(keyfield_values):
-            keyfield_values = keyfield_values[:max_experiments]
-        result_field_names = utils.get_result_field_names(self.config)
         try:
-            cpus = int(self.config['PY_EXPERIMENTER']['cpu.max'])
+            number_parallel_experiments = int(self.config['PY_EXPERIMENTER']['number_parallel_experiments'])
         except ValueError:
-            raise InvalidValuesInConfiguration('cpu.max must be an integer')
-        table_name = self.get_config_value('PY_EXPERIMENTER', 'table')
-        result_processors = [ResultProcessor(self.config, self.database_credential_file_path, table_name=table_name, condition=p,
-                                             result_fields=result_field_names) for p in keyfield_values]
-        experiment_functions = [experiment_function for _ in keyfield_values]
-        try:
-            custom_fields = [dict(self.config.items('CUSTOM')) for _ in keyfield_values]
-        except NoSectionError:
-            custom_fields = [None for _ in keyfield_values]
+            raise InvalidValuesInConfiguration('number_parallel_experiments must be an integer')
+
+        with Manager() as manager:
+            if max_number_experiments_to_execute != -1:
```

Comment:
It has to be `if max_number_experiments_to_execute > -1:`

Reply:
changed to >=0